File: apps/hotel-booking-service-admin/src/api/booking/Booking.ts
import { Customer } from "../customer/Customer";
import { Payment } from "../payment/Payment";
import { Room } from "../room/Room";

export type Booking = {
  checkInDate: Date | null;
  checkOutDate: Date | null;
  comment: string | null;
  createdAt: Date;
  customer?: Customer | null;
  description: string | null;
  id: string;
  name: string | null;
  payments?: Array<Payment>;
  room?: Room | null;
  totalPrice: number | null;
  updatedAt: Date;
};

File: apps/dotnet-service/src/APIs/Booking/Dtos/Booking.cs
namespace DotnetService.APIs.Dtos;

public class Booking
{
    public DateTime? CheckInDate { get; set; }

    public DateTime? CheckOutDate { get; set; }

    public string? Comment { get; set; }

    public DateTime CreatedAt { get; set; }

    public string? Customer { get; set; }

    public string? Description { get; set; }

    public string Id { get; set; }

    public string? Name { get; set; }

    public List<string>? Payments { get; set; }

    public string? Room { get; set; }

    public double? TotalPrice { get; set; }

    public DateTime UpdatedAt { get; set; }
}
